Products/ZenModel/migrate/addZproxyNginxImpactConfig.py

```diff
@@ -0,0 +1,85 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+
+__doc__ = """Add proper pagespeed filter for impact
+"""
+
+import logging
+import re
+log = logging.getLogger("zen.migrate")
+
+from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
+import Migrate
+import servicemigration as sm
+sm.require("1.1.5")
+
+org_statement='pagespeed Disallow \"*/static/*\";\n\n'
+new_statement='pagespeed  Disallow \"*/static/*\";\n        pagespeed  Disallow \"*/impact_graph*\";\n        pagespeed  Allow    \"*/impact_graph*.js\";\n\n'
+
+class AddZproxyNginxImpactConfig(Migrate.Step):
+    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION)
+
+    config_file = "/opt/zenoss/zproxy/conf/zproxy-nginx.conf"
+    save_file = "/opt/zenoss/var/ext/zproxy-nginx.conf.orig"
+
+    def update_config(self, zproxy):
+        commit, wrote = False, False
+        current_config, orig_config = None, None
+
+        for cfg in zproxy.configFiles:
+            if cfg.filename == self.config_file:
+                current_config = cfg
+                break
+
+        for cfg in zproxy.originalConfigs:
+            if cfg.filename == self.config_file:
+                orig_config = cfg
+                break
+
+        if orig_config:
+            new_content = orig_config.content.replace(org_statement, new_statement)
+            if orig_config != new_content:
```

Comment:
orig_config will never be the same as new_content, do you mean orig_config.content?

Reply:
If the migration already occurred or the orig_config was not of the expected pattern, then new_content will be the same as the orig_config.. In that case, we don't commit